### PR TITLE
Pipeline tests

### PR DESCRIPTION
**Pull request by kasei**
This adds testing which sanity check the expected output of the AATA pipeline. It also includes some refactoring to the AATA driver code to make it more amenable to testing.

One of the tests currently fails which is expected and represents a problem with how abstracts are currently handled. This will be fixed in a subsequent PR.

### CODE REVIEW

File: aata.py [hunk 1]
```diff
@@ -39,136 +39,162 @@
...
-		MakeLinkedArtLinguisticObject(),
-		AddDataDependentArchesModel(models=arches_models),
-		_input=records.output
-	)
-	if True:
-		# write ARTICLES data
-		graph.add_chain(
-			SRLZ,
-			WRITER,
+class AATAPipeline:
```
Comment (azaroth42) [inline aata.py]: In the future, this should be abstracted again to just Pipeline to manage cross-project functionality.  Which is not a blocker to merging now, of course.

File: aata.py [hunk 2]
```diff
@@ -39,136 +39,162 @@
...
+			sys.stderr.write('*** No serialization chain defined\n')
+
+	def add_articles_chain(self, graph, records):
+		if self.limit is not None:
+			records = graph.add_chain(
+				Limit(self.limit),
+				_input=records.output
+			)
+		articles = graph.add_chain(
+			make_aata_article_dict,
```
Comment (azaroth42) [inline aata.py]: If we're building a pipeline class per project, should the transformation functions be members of the class?

File: aata.py (hunk 1 above)
Comment (kasei) [reply aata.py]: Yeah. The changes I've got pending on top of this PR refactor this a bit, but definitely room for improvement.

File: aata.py (hunk 2 above)
Comment (kasei) [reply aata.py]: Possibly, though I'm wary of turning this into either one huge class for the entire project, or a deep class hierarchy. I'll try to circle back around after the code stabilizes a bit and see what could make sense being part of the project class.

**Pull request merged**